cpu/nrf51/periph/i2c.c

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2016 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_nrf51
+ * @{
+ *
+ * @file
+ * @brief       Low-level I2C driver implementation
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "mutex.h"
+#include "assert.h"
+#include "periph/i2c.h"
+#include "periph_conf.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#ifdef I2C_NUMOF
+
+/**
+ * @brief   If any of the 4 lower bits are set, the speed value is invalid
+ */
+#define INVALID_SPEED_MASK  (0x0f)
+
+/**
+ * @brief   Initialized bus locks (we have a maximum of two devices...)
+ */
+static mutex_t locks[] = {
+    MUTEX_INIT,
+    MUTEX_INIT
+};
+
+static inline NRF_TWI_Type *dev(i2c_t bus)
+{
+    return i2c_config[bus].dev;
+}
+
+static int error(i2c_t bus)
+{
+    dev(bus)->ERRORSRC = (TWI_ERRORSRC_DNACK_Clear |
+                          TWI_ERRORSRC_ANACK_Clear |
+                          TWI_ERRORSRC_OVERRUN_Clear);
+    dev(bus)->EVENTS_ERROR = 0;
+    DEBUG("[i2c] error 0x%02x\n", (int)dev(bus)->ERRORSRC);
```

Comment:
true - seems like I never used that output... moved it upwards to the top of the function.